src/engine/strat_engine/devlinks.rs

```diff
@@ -3,169 +3,97 @@
-    },
-    stratis::StratisResult,
+    engine::engine::DEV_PATH,
+    stratis::{ErrorEnum, StratisError, StratisResult},
 };
 
-/// Set up the root Stratis directory, where dev links as well as temporary
-/// MDV mounts will be created. This must occur before any pools are setup.
-pub fn setup_dev_path() -> StratisResult<()> {
-    if let Err(err) = fs::create_dir(DEV_PATH) {
-        if err.kind() != ErrorKind::AlreadyExists {
-            return Err(From::from(err));
-        }
-    }
-
-    Ok(())
-}
-
-/// Setup the pool directory and the symlinks in /stratis for the specified pool and filesystems
-/// it contains.
-// Don't just remove and recreate everything in case there are processes
-// (e.g. user shells) with the current working directory within the tree.
-pub fn setup_pool_devlinks(pool_name: &str, pool: &StratPool) {
-    if let Err(err) = || -> StratisResult<()> {
-        let pool_path = pool_directory(pool_name);
-
-        if !pool_path.exists() {
-            pool_added(pool_name);
-        }
+const UEVENT_PATH: &str = "/sys/class/block";
+const UEVENT_CHANGE_EVENT: &[u8] = b"change";
 
-        let mut existing_files = fs::read_dir(pool_path)?
-            .map(|dir_e| dir_e.map(|d| d.file_name().into_string().expect("Unix is utf-8")))
-            .collect::<Result<HashSet<_>, _>>()?;
-
-        for (fs_name, _, fs) in pool.filesystems() {
-            filesystem_added(pool_name, &fs_name, &fs.devnode());
-            existing_files.remove(&fs_name.to_owned());
-        }
-
-        for leftover in existing_files {
-            filesystem_removed(pool_name, &leftover);
-        }
-
-        Ok(())
-    }() {
-        warn!(
-            "setup_pool_devlinks failed for /stratis/{}, reason {:?}",
-            pool_name, err
-        );
-    };
+/// Given a pool name and a filesystem name, return the path it should be
+/// available as a device for mounting.
+pub fn filesystem_mount_path<T: AsRef<str>>(pool_name: T, fs_name: T) -> PathBuf {
+    vec![DEV_PATH, pool_name.as_ref(), fs_name.as_ref()]
+        .iter()
+        .collect()
 }
 
-/// Clean up directories and symlinks under /stratis based on current
-/// config. Clear out any directory or file that doesn't correspond to a pool.
-// Don't just remove everything in case there are processes
-// (e.g. user shells) with the current working directory within the tree.
-pub fn cleanup_devlinks<'a, I: Iterator<Item = (&'a Name, &'a PoolUuid, &'a StratPool)>>(pools: I) {
-    if let Err(err) = || -> StratisResult<()> {
-        let mut existing_dirs = fs::read_dir(DEV_PATH)?
-            .map(|dir_e| dir_e.map(|d| d.file_name().into_string().expect("Unix is utf-8")))
-            .collect::<Result<HashSet<_>, _>>()?;
-
-        for (pool_name, _, _) in pools {
-            existing_dirs.remove(&pool_name.to_owned());
-        }
-
-        for leftover in existing_dirs {
-            pool_removed(&Name::new(leftover));
+pub fn pool_renamed(pool_name: &str, _: &str) {
```

Comment:
At this point, the second parameter _can_ be dropped altogether.

Reply:
Resolved.